cadcetrans/cadcetrans/etrans_core.py

```diff
@@ -0,0 +1,756 @@
+        for f in rfiles.keys():
+            print('\t {:6} -'.format(f),
+                  colored('{:8d}'.format(len(rfiles[f])), 'red'))
+
+    print('\nTransferring files:')
+    tfiles = get_trans_files(dirname)
+    if not tfiles:
+        print('\tNone')
+    else:
+        errors = []
+        for f in tfiles:
+            if 'error' in f:
+                errors.append(f)
+            else:
+                print('\tpid {:7} - {:6} (started at {:15})'.
+                      format(f['pid'], f['files'], f['started']))
+        if errors:
+            print('\tErrors:')
+            for e in errors:
+                print(colored('\t\t{}'.format(e['error']), 'red'))
+
+    print('\nTransferred files*:')
+    lasth = [0, 0]
+    last24h = [0, 0]
+    last7d = [0, 0]
+    now = datetime.now()
+    for r in _get_transfer_log_info():
+        if 'put_cadc_file' in r:  # TODO - make it more robust
+            fields = r.split('put_cadc_file')
+            if len(fields) > 1:
+                logdate = datetime.strptime(fields[0].split('[')[0].strip(),
+                                            '%Y-%m-%d %H:%M:%S,%f')
+                timediff = now - logdate
+                index = 1
+                data = json.loads(fields[1].split('-')[1].strip())
+                if data['success']:
+                    index = 0
+                if timediff.total_seconds()/60.0/60.0 < 1:
+                    lasth[index] += 1
+                if timediff.total_seconds()/60.0/60.0 < 24:
+                    last24h[index] += 1
+                if timediff.total_seconds()/60.0/60.0 < 24 * 7:
+                    last7d[index] += 1
+                else:
+                    # TODO break reading the file here
+                    pass
+    print('\tLast hour success -', colored('{:6}'.format(lasth[0]), 'green'),
+          ' error -', colored('{:8d}'.format(lasth[1]), 'red'))
+    print('\tLast day  success -', colored('{:6}'.format(last24h[0]), 'green'),
+          ' error -', colored('{:8d}'.format(last24h[1]), 'red'))
+    print('\tLast week success -', colored('{:6}'.format(last7d[0]), 'green'),
+          ' error -', colored('{:8d}'.format(last7d[1]), 'red'))
+    logdir = etrans_config.get('etransfer', 'transfer_log_dir')
+    trans_log = os.path.join(logdir, 'cadcetrans.log')
+    print('* - details in {}'.format(trans_log))
+
+
+def update_backup(subject, dirname):
+    """
+    Logs an update of the status of processing and rejecting files and
+    transfers the latest logging files to the configured (vos) backup
+    :param subject:
+    :param dirname:
+    :return:
+    """
+    backup_dir = etrans_config.get('etransfer', 'backup_dir')
+    # this should come from the config instance
+    config_file = os.path.join(utils._CONFIG_PATH, 'cadc-etrans-config')
+    if not backup_dir:
+        raise RuntimeError('backup_dir must be specified in {}'.
+                           format(config_file))
+    if not backup_dir.startswith('vos:'):
```

Comment:
This runs at our clients - the backup to vospace is done so that Ling can keep an eye on what is going on especially when it comes to transfer errors which are caused by us most of the time.